src/Tests.cmake

```diff
@@ -1,9 +1,380 @@
 include_guard()
 
+include("${CMAKE_CURRENT_LIST_DIR}/SystemLink.cmake")
+
 # Enable coverage reporting for gcc/clang
 function(enable_coverage _project_name)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
     target_compile_options(${_project_name} INTERFACE --coverage -O0 -g)
     target_link_libraries(${_project_name} INTERFACE --coverage)
   endif()
 endfunction()
+
+function(_configure_target target_name type)
+  set(options)
+  set(one_value_args)
+  set(multi_value_args
+    SOURCES
+    INCLUDES
+    SYSTEM_INCLUDES
+    DEPENDENCIES_CONFIG
+    DEPENDENCIES
+    LIBRARIES
+    SYSTEM_LIBRARIES
+    COMPILE_DEFINITIONS
+    COMPILE_OPTIONS
+    COMPILE_FEATURES
+  )
+  cmake_parse_arguments(args "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  if(${type} STREQUAL "library_test")
+    add_executable(${target_name})
+    set(scope PRIVATE)
+  elseif(${type} STREQUAL "test_config")
+    add_library(${target_name} INTERFACE)
+    set(scope INTERFACE)
+  endif()
```

Comment:
What happens in the else condition?

Reply:
The function `_configure_target` currently only supports type `"library_test"` and `"test_config"`. Here the if-else checks the type to add the corresponding target, and set the `scope` correctly.

For instance, a `"library_test"` is an executable target, it will link all its libraries PRIVATE-ly; a `"test_config"` is an interface library as it is used to propagate the test configuration, and it will link its libraries libraries INTERFACE-ly.